plotly/tools.py

```diff
@@ -1443,6 +1443,376 @@ class FigureFactory(object):
+            for name in df:
+                if name != index:
+                    col_names.append(name)
+
+            # Populate matrix = []
+            for name in col_names:
+                matrix.append(df[name].values.tolist())
+
+            # Check if Matrix Values are either all strings or numbers
+            for vector in matrix:   
+                if ((type(vector[0]) == float) or (type(vector[0]) == int)):
+                    for entry in vector:
+                        if not ((type(entry) == float) or (type(entry) == int)):
+                            raise exceptions.PlotlyError("Error in data"
+                                                         "frame. Make sure "
+                                                         "that all entries "
+                                                         "of each column are "
+                                                         "either numbers or "
+                                                         "strings.")
+                                
+                if (type(vector[0]) == str):
+                    for entry in vector:
+                        if (type(entry) != str):
+                            raise exceptions.PlotlyError("Error in data"
+                                                         "frame. Make sure "
+                                                         "that all entries "
+                                                         "of each column are "
+                                                         "either numbers or "
+                                                         "strings.")
+
+            # Check if index_vals are either all strings or numbers 
+            if ((type(index_vals[0]) == float) or (type(index_vals[0]) == int)):
+                for entry in index_vals:
+                    if not ((type(entry) == float) or (type(entry) == int)):
+                        raise exceptions.PlotlyError("Error in data"
+                                                     "frame. Make sure "
+                                                     "that all entries "
+                                                     "of each column are "
+                                                     "either numbers or "
+                                                     "strings.")
+                            
+            if (type(index_vals[0]) == str):
+                for entry in index_vals:
+                    if (type(entry) != str):
+                        raise exceptions.PlotlyError("Error in data"
+                                                     "frame. Make sure "
+                                                     "that all entries "
+                                                     "of each column are "
+                                                     "either numbers or "
+                                                     "strings.")
```

Comment:
Lots of duplication here, you could write two functions `_is_string_vector` and `_is_number_vector` or something. Try and keep it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). :smile_cat: